Extensions/quick_tags.js

```diff
@@ -397,21 +399,26 @@ XKit.extensions.quick_tags = new Object({
 
 	},
 
-	do_posts: function() {
-
-		// get posts:
-		var posts = XKit.interface.get_posts("xkit-quick-tags-done", true, true);
-
-		$(posts).each(function() {
+	do_posts: async function() {
+		if (XKit.extensions.quick_tags.processing === true) {
+			return;
+		}
 
-			$(this).addClass("xkit-quick-tags-done");
+		if (XKit.interface.where().inbox) {
+			return;
+		}
 
-			if ($(this).hasClass("is_note") && XKit.interface.where().inbox === true) { return; }
+		XKit.extensions.quick_tags.processing = true;
+		var $posts = await XKit.interface.react.get_posts("xkit-quick-tags-done", true);
 
-			XKit.interface.add_control_button(this, "xkit-quick-tags", "");
+		$posts.each(function() {
+			const $post = $(this);
+			$post.addClass("xkit-quick-tags-done");
 
+			XKit.interface.react.add_control_button($post, "xkit-quick-tags", "");
```

Comment:
now that `$posts` is a jQuery set, we can add the class to the whole set at once rather than through iteration

```suggestion
		$posts
		.addClass("xkit-quick-tags-done")
		.each(function() {
			XKit.interface.react.add_control_button($(this), "xkit-quick-tags", "");
```